File: Blanquita-Backend/app/Models/BobinaPapel/BobinaPapel.py
from datetime import date
from decimal import Decimal
from typing import List, Optional
from pydantic import BaseModel


class BobinaPapel(BaseModel):
    IdBobinaPapel: Optional[int] = None
    CodigoBobina: str
    IdTipoBobina: int
    IdLoteBobina: int
    IdEstadoMateriaPrima: int
    PesoBrutoKg: Optional[Decimal] = None
    Gramaje: Optional[Decimal] = None
    PesoNetoKg: Optional[Decimal] = None


class ListaBobinasPapel(BaseModel):
    Bobinas: List[BobinaPapel]


class BobinaPapelIngresoItem(BaseModel):
    CodigoBobina: str
    PesoBrutoKg: Optional[Decimal] = None
    Gramaje: Optional[Decimal] = None
    PesoNetoKg: Optional[Decimal] = None


class IngresoModelo(BaseModel):
    IdProveedor: int
    IdTipoBobina: int
    Bobinas: List[BobinaPapelIngresoItem]


class IngresoLoteBobinaPapelResponse(BaseModel):
    FechaRecepcion: date
    CantidadBobinas: int



class TipoBobinaPapelIngreso(BaseModel):
    IdTipoBobina: int
    NombreTipoBobina: str

File: Blanquita-Frontend/src/models/BobinaPapel/BobinaPapel.js
export const BobinaPapel = (data) => ({
  IdBobinaPapel: data.IdBobinaPapel ?? null,
  CodigoBobina: data.CodigoBobina,
  IdTipoBobina: data.IdTipoBobina,
  IdLoteBobina: data.IdLoteBobina,
  IdEstadoMateriaPrima: data.IdEstadoMateriaPrima,
  PesoBrutoKg: data.PesoBrutoKg ?? null,
  Gramaje: data.Gramaje ?? null,
  PesoNetoKg: data.PesoNetoKg ?? null,
});

export const ListaBobinasPapel = (data) => ({
  Bobinas: (data.Bobinas ?? []).map(BobinaPapel),
});

export const BobinaPapelIngresoItem = (data) => ({
  CodigoBobina: data.CodigoBobina,
  PesoBrutoKg: data.PesoBrutoKg ?? null,
  Gramaje: data.Gramaje ?? null,
  PesoNetoKg: data.PesoNetoKg ?? null,
});

export const IngresoModelo = (data) => ({
  IdProveedor: data.IdProveedor,
  IdTipoBobina: data.IdTipoBobina,
  Bobinas: (data.Bobinas ?? []).map(BobinaPapelIngresoItem),
});

export const IngresoLoteBobinaPapelResponse = (data) => ({
  FechaRecepcion: data.FechaRecepcion,
  CantidadBobinas: data.CantidadBobinas,
});

export const TipoBobinaPapelIngreso = (data) => ({
  IdTipoBobina: data.IdTipoBobina,
  NombreTipoBobina: data.NombreTipoBobina,
});
